TypeSystem/flow-analysis/type_of_interest_A03_t01.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that at type `T` is a type of interest for a variable `x`
+/// in a set of tested types `tested` if `tested` contains a type `S` such that
+/// `T` is `S`, or `T` is `NonNull(S)`.
+///
+/// @description Checks that a type `void` cannot be made a type of interest for
+/// a variable.
```

Comment:
I think the variable has to have declared type `Object?`, or some other top type. Otherwise it could be allowed to make `void` a type of interest, but it should not make a difference because types of interest that aren't subtypes of the declared type of the variable can never have an effect on anything.

We shouldn't be able to promote the variable to have type `void`, because that isn't a _proper_ subtype of the type of the variable (promoted or not, it can't be a proper supertype of a top type).

Nevertheless, it does make sense to see that `if (s is Void) { /*..'s' could be `void` here..*/ }`, and the result should show that it does not have the type `void`, so (e.g.) `s.substring(0)` is OK.

Reply:
Removed top-types tests from this PR. I'll add them as promotion via assigment and promotion via type tests. And they will test type of interest as well.

Reply:
OK.